packages/destination-actions/src/destinations/metronome/sendEvent/__tests__/index.test.ts

```diff
@@ -0,0 +1,101 @@
+import nock from 'nock'
+import { createTestEvent, createTestIntegration } from '@segment/actions-core'
+import Destination from '../../index'
+
+const testDestination = createTestIntegration(Destination)
+
+describe('Metronome.sendEvent', () => {
+  describe("process", () => {
+    it('should send an event', async () => {
+      const event = createTestEvent();
+
+      nock('https://api.getmetronome.com').post('/v1/ingest').reply(200)
+
+      const responses = await testDestination.testAction('sendEvent', {
+        event,
+        useDefaultMappings: true,
+        settings: {
+          apiToken: "mock-api-token",
+        },
+        mapping: {
+          event_type: { '@path': "$.event" },
+          properties: { '@path': "$.context" },
+        },
+      });
+
+      expect(responses.length).toBe(1)
+      expect(responses[0].status).toBe(200)
+
+      // TODO: Feels like there should be a lighter weight to do this check
+      expect(responses[0].options.headers).toMatchInlineSnapshot(`
+        Headers {
+          Symbol(map): Object {
+            "authorization": Array [
+              "Bearer mock-api-token",
+            ],
+            "user-agent": Array [
+              "Segment (Actions)",
+            ],
+          },
+        }
+      `)
+
+      expect(responses[0].options.json).toEqual([
+        {
+          event_type: event.event,
+          customer_id: event.groupId || event.userId || event.anonymousId,
+          properties: event.context,
+          transaction_id: event.messageId,
+          timestamp: new Date(event.timestamp ?? "").toISOString(),
+        }
+      ])
+    })
+
+
+    it('should convert a non rfc3999 timestamp', async () => {
```

Comment:
can you also include a test checking we convert epoch timestamps properly too?

Reply:
4c435e8